base/abstractarray.jl

```diff
@@ -803,9 +825,8 @@ typed_hcat(T::Type, A::AbstractArray...) = cat_t(2, T, A...)
 function hvcat(nbc::Integer, as...)
     # nbc = # of block columns
     n = length(as)
-    if mod(n,nbc) != 0
-        throw(ArgumentError("all rows must have the same number of block columns"))
-    end
+    mod(n,nbc) != 0 &&
+        throw(ArgumentError("all rows must have the same number of block columns ($nbc)"))
```

Comment:
As above, if you want to make the message even more useful, you could print the actual lengths of the rows that were passed.